### PR TITLE
fix(API): use `hexutil.Big` for `l1Fee` in `GetTransactionReceipt`

### PR DESCRIPTION
**Pull request by mask-pp**
## 1. Purpose or design rationale of this PR
~~The `zero address` issue has found. It because scroll-v3.1.4 use `*big.Int` but scroll-v3.1.12 use `*hexutil.Big`.~~
~~`*big.Int` marshal result is `0` but can't unmarshal `"0x0"`.~~
~~`*hexutil.Big` marshal result is `"0x0"` but can't unmarshal `0`.~~

~~So if we use `L1Fee             *hexutil.Big` unmarshal it, so the type should be `*hexutil.Big` but not `hexutil.Uint64`.~~

Use `hexutil.Big` to encode `receipt.l1Fee` to avoid overflow during conversion.

## 2. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [x] Yes

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -1669,7 +1669,7 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 		"logs":              receipt.Logs,
 		"logsBloom":         receipt.Bloom,
 		"type":              hexutil.Uint(tx.Type()),
-		"l1Fee":             hexutil.Uint64(receipt.L1Fee.Uint64()),
+		"l1Fee":             (*hexutil.Big)(receipt.L1Fee),
```
Comment (Thegaram) [inline internal/ethapi/api.go:1672]: `blockNumber`, `transactionIndex`, etc all use `hexutil.Uint64`, so let's keep `l1Fee` consistent with them.

Comment (mask-pp) [reply internal/ethapi/api.go:1672]: But in [marshal and unmarshal](https://github.com/scroll-tech/go-ethereum/blob/910d4f1a70fc29d4433b9c793191e29203489d9d/core/types/gen_receipt_json.go#L68) interface we use `*hexutil.Big` type, this is inconsistent.

Comment (Thegaram) [reply internal/ethapi/api.go:1672]: `BlockNumber` in `UnmarshalJSON` also uses `*hexutil.Big`, does that also lead to issues?

Comment (mask-pp) [reply internal/ethapi/api.go:1672]: It's not a bug, this PR is just a suggested modification.

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -741,10 +741,10 @@ func (s *PublicBlockChainAPI) GetHeaderByHash(ctx context.Context, hash common.H
...
 
 // GetBlockByNumber returns the requested canonical block.
-// * When blockNr is -1 the chain head is returned.
-// * When blockNr is -2 the pending chain head is returned.
-// * When fullTx is true all transactions in the block are returned, otherwise
-//   only the transaction hash is returned.
+//   - When blockNr is -1 the chain head is returned.
+//   - When blockNr is -2 the pending chain head is returned.
+//   - When fullTx is true all transactions in the block are returned, otherwise
+//     only the transaction hash is returned.
```
Comment (0xmountaintop) [inline internal/ethapi/api.go]: can we avoid formatting. otherwise it will increase the auditing overhead&cost

Comment (mask-pp) [reply internal/ethapi/api.go]: ok

File: internal/ethapi/api.go (hunk 1 above)
Comment (colinlyguo) [reply internal/ethapi/api.go:1672]: What about changing "blockNumber" too?

Comment (mask-pp) [reply internal/ethapi/api.go:1672]: `BlockNumber`'s type is uint64 in `GetTransactionReceipt`, and the user-facing interface is also of type uint64, so it's no need to change the type again.
But `l1Fee`'s user-facing interface is `*big.Int` type, so keeping it consistent will be better.

**Review by colinlyguo — DISMISSED** [review]

File: internal/ethapi/api.go (hunk 1 above)
Comment (Thegaram) [reply internal/ethapi/api.go:1672]: I'd prefer to keep it consistent with how geth handles other fields.

**Comment by 0xmountaintop** [comment]
update version then looks good

File: internal/ethapi/api.go (hunk 1 above)
Comment (Thegaram) [reply internal/ethapi/api.go:1672]: You're right, it's better to use `*hexutil.Big` to avoid overflow.

**Comment by Thegaram** [comment]
Could you please share an example output of this API after this change? Is it like `"l1Fee": "0x1234"`?

**Comment by mask-pp** [comment]
> Could you please share an example output of this API after this change? Is it like `"l1Fee": "0x1234"`?

Ok, I will add it.

**Comment by 0xmountaintop** [comment]
> > Could you please share an example output of this API after this change? Is it like `"l1Fee": "0x1234"`?
> 
> Ok, I will add it.

add both "before" vs "after"

**Comment by mask-pp** [comment]
> 

before:
`"l1Fee": "0x0",`

after:
`"l1Fee": "0x0",`

They are same because `hexutil.Uint64` and `hexutil.Big` are both use hexadecimal strings as marshal result.

**Review by Thegaram — DISMISSED** [review]

**Review by Thegaram — APPROVED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Pull request merged**